vector/src/main/java/im/vector/app/core/extensions/BottomSheetDialog.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.extensions
+
+import android.util.DisplayMetrics
+import androidx.annotation.FloatRange
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+@Suppress("DEPRECATION")
```

Comment:
`WindowManager. defaultDisplay` and `getMetrics` are deprecated in 30. Could you add API 30 implementation? And also, please, use suppress on a line, where it's needed, not on whole function